presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergTableHandle.java

```diff
@@ -33,6 +33,7 @@ public class IcebergTableHandle
     private final TupleDomain<IcebergColumnHandle> predicate;
     private final boolean snapshotSpecified;
     private final Optional<String> tableSchemaJson;
+    private final Optional<Integer> partitionSpecId;
```

Comment:
Due to renaming partition column would cause partition spec evolution, maybe the `partitionSpecId` here should be `Optional<Set<Integer>>` or something like this. For renaming partition column would not change the partition column id and it's source column id in schema, but it would create a new partition spec.

Reply:
Another great point! Didn't think about renaming, changed to use partition field ids.